### DISCUSSION TITLE
ssh compression option

### DISCUSSION THREAD

**Discussion by 0deuce0** [Eugeny/tabby#5660] [Q&A]
Do we have the option to enable compression on ssh?   

-C

     -C      Requests compression of all data (including stdin, stdout, stderr, and data for forwarded X11, 
     TCP and UNIX-domain connections). The compression algorithm is the same used by gzip(1).  
     Compression is desirable on modem lines and other slow connections, but will only slow down 
     things on fast networks.  The default value can be set on a host-by-host basis in the configuration 
     files; see the Compression option.

**Comment by tv21** [comment]
I came here to ask specifically about this and found this issue.  As a workaround, when creating the new profile, instead of starting with an SSH connection you can scroll down to the ">_  bash /bin/bash" or similar (whatever shell your system uses by default) and then enter the ssh command as you would if you were at a normal command prompt; I did this with one connection so I could use autossh.  Be use to enter the full path to ssh, in other words

/usr/bin/ssh -C .....

(assuming ssh is located in the /usr/bin directory, you can use "which ssh" from a command prompt to be certain).

But I agree that the compression option is something that preferably ought to be built into Tabby, so that people who are on ISPs with usage caps and/or are transferring large amounts of data can gain the benefits of compression.

**Comment by Robertas-M** [comment]
+1 for this feature. It really makes sense with X forwarding - event with good connectivity, performance is improved noticeably with compression enabled
